File: Python3/Binary Search/2300. Successful Pairs of Spells and Potions..py
""" 2300. Successful Pairs of Spells and Potions
Difficulty: Medium
Submisson Link: https://leetcode.com/problems/successful-pairs-of-spells-and-potions/submissions/1376870229/ """

class Solution:
    def successfulPairs(self, spells: List[int], potions: List[int], success: int) -> List[int]:
        n, _ = len(potions), potions.sort()
        return [n - bisect_left(potions, success / s) for s in spells]

File: C++/Binary Search/2300. Successful Pairs of Spells and Potions.cpp
/* 2300. Successful Pairs of Spells and Potions
Difficulty: Medium
Submisson Link: https://leetcode.com/problems/successful-pairs-of-spells-and-potions/submissions/1376870711/ */

class Solution {
public:
    vector<int> successfulPairs(vector<int>& spells, vector<int>& potions, long long success) {
        
        int n = spells.size();
        int m = potions.size();
        vector<int> pairs;

        sort(potions.begin(), potions.end());

        for(int i = 0; i < n; i++)
            pairs.push_back(m - (lower_bound(potions.begin(), potions.end(), (double)success / spells[i]) - potions.begin()));

        return pairs;
    }
};